### PR TITLE
fix(drivers): retry transient model provider errors instead of crashing

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

When a model provider (e.g. Minimax via OpenRouter) returns a transient error like "invalid function arguments", the workflow now retries with exponential backoff instead of failing permanently. This introduces a `ModelProviderError` exception type to distinguish provider-side issues from Amelia validation errors.

### Added
- `ModelProviderError` exception in `amelia/core/exceptions.py` to distinguish provider errors from Amelia validation errors
- Provider error detection and wrapping in `ApiDriver.generate()` and `ApiDriver.execute_agentic()`
- `ModelProviderError` to `TRANSIENT_EXCEPTIONS` for automatic retry
- Retry handling in `approve_workflow()` for post-approval transient failures
- `AMELIA_PROVIDER_ERROR_PATTERNS` env var for configurable error pattern matching
- LRU-cached `_get_provider_error_patterns()` for efficient pattern lookup
- Comprehensive unit tests for provider error detection, wrapping, exponential backoff, and retry logic

### Changed
- Exponential backoff capped at `2^31` to prevent integer overflow on large attempt counts
- Inlined `REQUIRED_AGENTS` constant in `config.py` (removed from `types.py`)
- Removed warning log from `Profile.get_agent_config()` (still raises `ValueError`)

### Removed
- `KnowledgeService` and `IngestionPipeline` (unused knowledge library modules)
- `_summarize_stage_output` and `_truncate_nested` helpers (moved out in prior work)
- Stale test files for removed modules

## Motivation

Provider-side errors (bad JSON from LLM, streaming failures, provider 400s) were surfacing as unhandled `ValueError`s, crashing workflows that could succeed on retry. This is especially common with OpenRouter providers like Minimax that occasionally return malformed tool call arguments.

## Testing

- [x] Unit tests added for `ModelProviderError`, `_is_model_provider_error`, `_extract_provider_info`
- [x] Unit tests for exponential backoff edge cases (overflow, max delay capping, retry exhaustion)
- [x] Unit tests for `approve_workflow` retry loop with transient errors
- [x] Tests for custom provider error patterns via `AMELIA_PROVIDER_ERROR_PATTERNS` env var

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/unit/server/orchestrator/test_service.py [hunk 1]
```diff
@@ -1318,3 +1321,297 @@ async def test_task_title_defaults_description_to_title(
...
+        patch.object(
+            orchestrator, "_create_server_graph", return_value=setup.mock_graph
+        ),
+        patch.object(orchestrator, "_resolve_prompts", return_value={}),
+        patch.object(orchestrator, "_emit", new=AsyncMock()),
+        patch(
+            "amelia.server.orchestrator.service.asyncio.sleep", new_callable=AsyncMock
+        ) as mock_sleep,
+    ):
+        yield mock_sleep
```
[the remark this answers is not in the document]
Comment (anderskev) [reply tests/unit/server/orchestrator/test_service.py]: @coderabbitai Fixed in `2502654e`. Replaced MagicMock with real `ServerExecutionState`. This also caught that the mock had `.workflow_id` while the real model uses `.id` — the mismatch was masked by MagicMock.
